### PR TITLE
Introduce optional components

### PR DESCRIPTION
**Pull request by kvaps**
This PR introduces an extra option to cozystack configmap. It allows to enable components that are optional to specific bundle name, example usage:

```yaml
bundle-enable: telepresence,external-dns,external-secrets-operator
```

## Summary by CodeRabbit

- **New Features**
	- Added optional attributes to several release entries, allowing for more flexible deployment configurations.
	- Enhanced Helm release generation logic to consider both enabled and disabled states for components.
	- Improved namespace management by incorporating checks for optional components based on enablement and disablement.

- **Bug Fixes**
	- Updated dependencies for `external-dns` to include `cilium` and `kubeovn`.

- **Documentation**
	- Clarified the configurability of deployment components with the introduction of optional attributes.

### CODE REVIEW

**Review by mrkhachaturov — COMMENTED** [review]
To disable specific components within a bundle, utilize the `bundle-disable` option by supplying a comma-separated list of the components you wish to deactivate.

It seems that for optional components, a repetitive approach is used, where some components are disabled by default and need to be enabled. Perhaps we could consider a more streamlined method for managing this activation process.

To maintain clarity within our bundles, we might consider creating additional bundles that function as add-ons. This approach would allow us to incorporate specific components through a configuration map, enabling users to seamlessly define and integrate extra add-ons into their existing bundles.

Let me know what you think about creating a new bundle (add-on) where we can place components that can be optionally added to the main bundles via a ConfigMap.

File: packages/core/platform/bundles/paas-full.yaml [hunk 1]
```diff
@@ -215,16 +215,19 @@ releases:
   chart: cozy-capi-providers
   namespace: cozy-cluster-api
   privileged: true
+  optional: true
```
Comment (themoriarti) [inline packages/core/platform/bundles/paas-full.yaml]: How to understand it? if there is no cluster api how will it work?

Comment (kvaps) [reply packages/core/platform/bundles/paas-full.yaml]: Nice catch, thank you, that was intended for `telepresence` chart not `cozy-capi-providers`

**Comment by kvaps** [comment]
> In general, some kind of overengineering feature, it is not very clear what it is for, if you can simply disable this or that module within the bundle.

This feature is intended only for optional components, now `telepresence`, `external-dns`, `external-secrets-operator` charts will not be enabled by default.

**Review by themoriarti — COMMENTED** [review]
Now looks good.

**Pull request merged**